Need to figure out publishing plugin incompaibility with 6.0.0 shadow plugin

with version

```diff
git diff
diff --git a/build.gradle b/build.gradle
index 25fbe6ce..5120231f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -19,7 +19,7 @@
 /***************************/
 
 plugins {
-  id "com.github.johnrengelman.shadow" version "5.2.0"
+  id "com.github.johnrengelman.shadow" version "6.0.0"
   id "com.github.spotbugs" version "4.5.0"
   id "io.codearte.nexus-staging" version "0.22.0"
   id "de.marcphilipp.nexus-publish" version "0.4.0"
```

```
~  ./gradlew publishMavenJavaPublicationToMavenLocal
> Task :api:publishMavenJavaPublicationToMavenLocal FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':api:publishMavenJavaPublicationToMavenLocal'.
> Failed to publish publication 'mavenJava' to repository 'mavenLocal'
   > Invalid publication 'mavenJava': multiple artifacts with the identical extension and classifier ('jar', 'all').

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.6.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 8s
9 actionable tasks: 8 executed, 1 up-to-date
```

maven publication fails, whereas with 5.2.0 it works fine
```
~  ./gradlew publishMavenJavaPublicationToMavenLocal

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.6.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 2s
9 actionable tasks: 4 executed, 5 up-to-date
```

--- a/build.gradle
+++ b/build.gradle
@@ -179,6 +179,10 @@ project(':api') {
         }
     }
 
+    artifacts {
+        archives javadocJar, sourcesJar, shadowJar
+    }
+
     publishing {
         publications {
             mavenJava(MavenPublication) {

--- a/build.gradle
+++ b/build.gradle
@@ -19,7 +19,7 @@
 /***************************/
 
 plugins {
-  id "com.github.johnrengelman.shadow" version "5.2.0"
+  id "com.github.johnrengelman.shadow" version "6.1.0"
   id "com.github.spotbugs" version "4.5.0"
   id "io.codearte.nexus-staging" version "0.22.0"
   id "de.marcphilipp.nexus-publish" version "0.4.0"
@@ -156,14 +156,9 @@ project(':api') {
         }
     }
 
-    task sourcesJar(type: Jar) {
-        classifier = 'sources'
-        from sourceSets.main.allJava
-    }
-
-    task javadocJar(type: Jar, dependsOn: javadoc) {
-        classifier = 'javadoc'
-        from javadoc.destinationDir
+    java {
+        withJavadocJar()
+        withSourcesJar()
     }
 
     javadoc.options {
@@ -184,18 +179,11 @@ project(':api') {
         }
     }
 
-    artifacts {
-        archives javadocJar, sourcesJar, shadowJar
-    }
-
     publishing {
         publications {
             mavenJava(MavenPublication) {
                 artifactId archivesBaseName
                 from components.java
-                artifact sourcesJar
-                artifact shadowJar
-                artifact javadocJar
                 pom {
                     name = 'minio'
                     packaging = 'jar'

